map.go

```diff
@@ -311,10 +311,15 @@ func createMap(spec *MapSpec, inner *internal.FD, opts MapOptions, handles *hand
 		}
 	}
 
-	if spec.Flags&(unix.BPF_F_RDONLY_PROG|unix.BPF_F_WRONLY_PROG) > 0 || spec.Freeze {
+	switch {
```

Comment:
I don't think rdonly and mmapable flags are mutually exclusive. Selftest `test_mmap.c` contains a map with flags `BPF_F_MMAPABLE | BPF_F_RDONLY_PROG`, so we should probably always execute both checks here.

Reply:
Ah, good catch, fixed!